src/service/routing/spec.clj

```diff
@@ -29,9 +31,7 @@
 (s/def ::total spec/int?)
 (s/def ::total-map (s/keys :req-un [::total]))
 
-(s/def ::direction-map (s/keys :req-un [::direction]))
-(s/def ::direction (s/keys :req-un [::code ::waypoints ::routes]
-                           :opt []))
+(s/def ::direction (s/nilable (s/keys :req-un [::code ::waypoints ::routes])))
```

Comment:
Since we want to conform to Mapbox API, here we have to always return a not-nil response with corresponding code in it.